portal-frontend/src/app/features/notice-of-intents/alcs-edit-submission/alcs-edit-submission.component.ts

```diff
@@ -0,0 +1,230 @@
+    if (!this.noiSubmission) {
+      this.overlayService.showSpinner();
+      this.noiSubmission = await this.noticeOfIntentSubmissionDraftService.getByFileId(fileId);
+      this.loadOriginalSubmission(fileId);
+      const documents = await this.noticeOfIntentDocumentService.getByFileId(fileId);
+      if (documents) {
+        this.$noiDocuments.next(documents);
+      }
+      this.fileId = fileId;
+      this.$noiSubmission.next(this.noiSubmission);
+      this.overlayService.hideSpinner();
+    }
+  }
+
+  // this gets fired whenever applicant navigates away from edit page
+  async canDeactivate(): Promise<Observable<boolean>> {
+    await this.saveSubmission(this.customStepper.selectedIndex);
+    return of(true);
+  }
+
+  async onStepChange($event: StepperSelectionEvent) {
+    // scrolls to step if step selected programmatically
+    scrollToElement({ id: `stepWrapper_${$event.selectedIndex}`, center: false });
+  }
+
+  async saveSubmission(step: number) {
+    switch (step) {
+      case EditNoiSteps.Parcel:
+        await this.parcelDetailsComponent.onSave();
+        break;
+      case EditNoiSteps.PrimaryContact:
+        await this.primaryContactComponent.onSave();
+        break;
+      case EditNoiSteps.Government:
+        await this.selectGovernmentComponent.onSave();
+        break;
+      case EditNoiSteps.LandUse:
+        await this.landUseComponent.onSave();
+        break;
+      case EditNoiSteps.Attachments:
+        await this.otherAttachmentsComponent.onSave();
+        break;
+      case EditNoiSteps.Proposal:
+        if (this.rosoProposalComponent) {
+          await this.rosoProposalComponent.onSave();
+        }
+        if (this.pofoProposalComponent) {
+          await this.pofoProposalComponent.onSave();
+        }
+        if (this.pfrsProposalComponent) {
+          await this.pfrsProposalComponent.onSave();
+        }
+        break;
+      case EditNoiSteps.ExtraInfo:
+        if (this.rosoAdditionalInfoComponent) {
+          await this.rosoAdditionalInfoComponent.onSave();
+        }
+        break;
+      case EditNoiSteps.ReviewAndSubmit:
+        //DO NOTHING
+        break;
+      default:
+        this.toastService.showErrorToast('Error updating notice of intent.');
+    }
+  }
+
+  async switchStep(index: number) {
+    // navigation to url will cause step change based on the index (index starts from 0)
+    // The save will be triggered using canDeactivate guard
+    this.showValidationErrors = this.customStepper.selectedIndex === EditNoiSteps.ReviewAndSubmit;
+    await this.router.navigateByUrl(`/alcs/notice-of-intent/${this.fileId}/edit/${index}`);
+  }
+
+  onParcelDetailsInitialized() {
+    if (this.expandedParcelUuid && this.parcelDetailsComponent) {
+      this.parcelDetailsComponent.openParcel(this.expandedParcelUuid);
+      this.expandedParcelUuid = undefined;
+    }
+  }
+
+  async onDownloadPdf(fileNumber: string | undefined) {
+    //TODO: Add this
```

Comment:
nitpick: I would suggest adding toastService.error('Not Implemented') or similar to avoid additional questions during testing